src/core_plugins/kibana/server/lib/supported_features/get_supported_features.js

```diff
@@ -0,0 +1,25 @@
+import { checkFieldStatsApi } from './feature_checks';
+
+
+const featureChecks = [
+  checkFieldStatsApi,
+];
+
+export async function getSupportedFeatures(callWithRequest) {
+  const { nodes: nodesInfo } = await callWithRequest('nodes.info', {
+    filterPath: [
+      'nodes.*.name',
+      'nodes.*.version',
+    ],
+  });
+
+  const clusterInfo = {
+    nodes: nodesInfo,
+  };
+
+  const featureFlags = featureChecks.reduce((featureFlags, featureCheck) => (
```

Comment:
By using `reduce` we require all `featureChecks` functions to attach the previous reduced value to their returning object again (`field_stats_ap.js:16`). Since all features checks should (imho) check for different independent features, I would suggest, making `featureChecks` an object, with the feature id (e.g. `field_stats_api`) as a key and the function attached as the value. I would then iterate over all keys here, and call the appropriate function, which would then only need to return an object `{ supported, reasons }` which will be merged into one object here in this place. Advantages: the feature check functions, wouldn't need to carry on the previous feature check results; a single place to see the feature ids, that API may return; each feature check is completely decoupled from the rest; and (maybe) a slightly better code readability, instead of passing a reduce function further down other files (but that's really just a minor improvement).